code/game/machinery/doppler_array.dm

```diff
@@ -1,5 +1,8 @@
 GLOBAL_LIST_EMPTY(doppler_arrays)
 
+#define RESEARCH	"research"
+#define CREDIT	"credit"

```

Comment:
think about moving these to __DEFINES/research.dm and maybe renaming them to "TACHYON_MODE_RESEARCH" and "TACHYON_MODE_CREDIT" to avoid collisions